src/webgpu/api/operation/texture_view/format_reinterpretation.spec.ts

```diff
@@ -0,0 +1,390 @@
+              return textureLoad(src, vec2<i32>(coord.xy), 0);
+            }`,
+            }),
+      entryPoint: 'main',
+      targets: [{ format }],
+    },
+    multisample: {
+      count: multisample.render,
+    },
+  });
+}
+
+g.test('texture_binding')
+  .desc(`Test that a regular texture allocated as 'format' is correctly sampled as 'viewFormat'.`)
+  .params(u =>
+    u //
+      .combine('format', kRegularTextureFormats)
+      .combine('viewFormat', kRegularTextureFormats)
+      .filter(
+        ({ format, viewFormat }) => format !== viewFormat && viewCompatible(format, viewFormat)
+      )
+  )
+  .fn(async t => {
+    const { format, viewFormat } = t.params;
+
+    // Make an input texel view.
+    const inputTexelView = makeInputTexelView(format);
+
+    // Write data out to bytes.
+    const texelSize = kTextureFormatInfo[format].bytesPerBlock;
+    const textureData = new Uint8ClampedArray(texelSize * kTextureSize * kTextureSize);
+    inputTexelView.writeTextureData(textureData, {
+      bytesPerRow: texelSize * kTextureSize,
+      rowsPerImage: kTextureSize,
+      subrectOrigin: [0, 0],
+      subrectSize: [kTextureSize, kTextureSize],
+    });
+
+    // Create and upload data to the texture.
+    const texture = t.device.createTexture({
+      format,
+      size: [kTextureSize, kTextureSize],
+      usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.TEXTURE_BINDING,
+      viewFormats: [viewFormat],
+    });
+
+    t.device.queue.writeTexture(
+      { texture },
+      textureData,
+      {
+        bytesPerRow: texelSize * kTextureSize,
+      },
+      [kTextureSize, kTextureSize]
+    );
+
+    // Reinterepret the texture as the view format.
+    // Make a texel view of the format that also reinterprets the data.
+    const reinterpretedView = texture.createView({ format: viewFormat });
+    const reinterpretedTexelView = TexelView.fromTexelsAsBytes(viewFormat, inputTexelView.bytes);
+
+    // Create a pipeline to write data out to rgba8unorm.
+    const pipeline = t.device.createComputePipeline({
+      compute: {
+        module: t.device.createShaderModule({
+          code: `
+          @group(0) @binding(0) var src: texture_2d<f32>;
+          @group(0) @binding(1) var dst: texture_storage_2d<rgba8unorm, write>;
+          @stage(compute) @workgroup_size(1, 1) fn main(
+            @builtin(global_invocation_id) global_id: vec3<u32>,
+          ) {
+            var coord = vec2<i32>(global_id.xy);
+            textureStore(dst, coord, textureLoad(src, coord, 0));
+          }`,
+        }),
+        entryPoint: 'main',
+      },
+    });
+
+    // Create an rgba8unorm output texture.
+    const outputTexture = t.device.createTexture({
+      format: 'rgba8unorm',
+      size: [kTextureSize, kTextureSize],
+      usage: GPUTextureUsage.STORAGE_BINDING | GPUTextureUsage.COPY_SRC,
+    });
+
+    // Execute a compute pass to load data from the reinterpreted view and
+    // write out to the rgba8unorm texture.
```

Comment:
Use rgba32float as the output format? Has better precision and range - not that it matters in practice right now.

Reply:
It has better precision but then we need to convert our ULP difference of 1 or 2 in unorm format to float. I think that makes the test less clear.

Reply:
That's something to think about - you have one format but need ULPs in terms of another format.